src/bokeh/models/annotations/legends.py

```diff
@@ -581,7 +607,19 @@ def _check_non_matching_scale_bar_unit(self):
     """)
 
     bar_length = NonNegative(Either(Float, Int))(default=0.2, help="""
-    The length of the bar, either a fraction of the frame or a number of pixels.
+    The length of the bar.
+
+    This is either a fraction of the frame, a number of pixels or
+    distance in the data space, depending on the configuration of
+    ``bar_length_units``.
+    """)
+
+    bar_length_units = Enum("screen", "data", default="screen", help="""
+    Defines how to interpret ``bar_length``.
+
+    Supported values are:
+    * ``"screen"`` - the length is either pixels or percentage of the frame
+    * ``"data"`` - the length is provided in data units
```

Comment:
I think it need to be explicit in the "screen" case that it is fractional values that are interpreted as a percentage and integral values that interpreted as pixel distances. Or else split those cases entirely and have a "percent" unit.

Reply:
I will add `"percent"` unit, so that there's no ambiguity.

Reply:
Actually, I will add `"percent"` for disambiguation, but I have to keep the original logic for backwards compatibility. I will add a note to the `"screen"` case about percentage value range.